### PR TITLE
Update courses discovery WordPress data loader for subjects translations

### PR DESCRIPTION
**Pull request by etechdevai**
**Description:**  This PR updates courses discovery WordPress data loader for subjects translations.

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-3322

**Merge checklist:**

- [ ] All reviewers approved
- [ ] Commits are (reasonably) squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** Please add screenshots for any viewable change.

### CODE REVIEW

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 1]
```diff
@@ -1016,6 +1016,15 @@ def _add_course_subjects(self, categories, course_run):
                 subject.slug = category['slug']
                 subject.save()
 
+            if subject:
+                for language_code, translated_title in category['title_translations'].items():
+                    subject_translation, __ = SubjectTranslation.objects.get_or_create(
+                        master_id=subject.pk,
+                        language_code=language_code
+                    )
```
Comment (taimoor-ahmed-1) [inline course_discovery/apps/course_metadata/data_loaders/api.py:1024]: nit: add empty line.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 2]
```diff
@@ -1016,6 +1016,15 @@ def _add_course_subjects(self, categories, course_run):
...
+                    subject_translation.name = translated_title if translated_title else category['title']
```
Comment (taimoor-ahmed-1) [inline course_discovery/apps/course_metadata/data_loaders/api.py:1025]: also, we should check if `title` is present or use `get` method

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 3]
```diff
@@ -1016,6 +1016,15 @@ def _add_course_subjects(self, categories, course_run):
                 subject.slug = category['slug']
                 subject.save()
 
+            if subject:
+                for language_code, translated_title in category['title_translations'].items():
```
Comment (taimoor-ahmed-1) [inline course_discovery/apps/course_metadata/data_loaders/api.py:1020]: will `title_translations` key always be present in `category` dict?  we could use `get` method.

File: course_discovery/apps/course_metadata/data_loaders/api.py (hunk 2 above)
Comment (etechdevai) [reply course_discovery/apps/course_metadata/data_loaders/api.py:1025]: @taimoor-ahmed-1 translated_title is optional while a category cannot be created without title hence we don't need to check if title is present or not.

File: course_discovery/apps/course_metadata/data_loaders/api.py (hunk 1 above)
Comment (etechdevai) [reply course_discovery/apps/course_metadata/data_loaders/api.py:1024]: We avoid unnecessary empty lines in code.

File: course_discovery/apps/course_metadata/data_loaders/api.py (hunk 3 above)
Comment (etechdevai) [reply course_discovery/apps/course_metadata/data_loaders/api.py:1020]: Done

**Review by taimoor-ahmed-1 — APPROVED** [review]
👍

**Review by umerabbas — APPROVED** [review]
👍🏻

**Pull request merged**